src/status_im2/contexts/chat/events.cljs

```diff
@@ -282,16 +289,15 @@
                               :on-error   #(log/error "failed to clear history " chat-id %)}]}
             (clear-history chat-id remove-chat?)))
 
-(rf/defn remove-chat
-  "Removes chat completely from app, producing all necessary effects for that"
-  {:events [:chat.ui/remove-chat]}
+(rf/defn close-and-remove-chat
+  "Closes the chat and removes it from chat list while retaining history, producing all necessary effects for that"
+  {:events [:chat.ui/close-chat]}
   [{:keys [db now] :as cofx} chat-id]
   (rf/merge cofx
             {:effects/push-notifications-clear-message-notifications [chat-id]
              :dispatch                                               [:shell/close-switcher-card
                                                                       chat-id]}
-            (deactivate-chat chat-id)
-            (offload-messages chat-id)))
+            (remove-chat chat-id)))
```

Comment:
could you elaborate on this change, it looks pretty critical

Reply:
The requirement for the destructive actions in the context drawer for 1:1 chat was changed. One of the actions was a delete message, but it was changed to close message where the message is just removed from the chat list while still retaining history.

This was communicated here: https://discord.com/channels/624634427930312714/928625369542713396/1167063940816179271

Reply:
mhm i think we already had something similar cc @churik

Reply:
and we have like `active?` flag or something

Reply:
🤔 The current implementation deleted the history alongside removing it from the home chat

Reply:
i don't see it removes history, it just clear it in react, we still have it in status-go

Reply:
I don't see the history after I delete it and try chatting with the other device again, even after closing the app and re-opening 🤔

Reply:
so there are two parts where we have data, react and status-go, in your PR as soon as you relogin chat will be back, im not sure that is expected, so i would say current version in develop, works as expected

Reply:
I am not sure I understand, here's my experience with `develop`:

https://github.com/status-im/status-mobile/assets/45393944/c560e056-f706-4b39-b0eb-d51846fcf900

I don't get to see the messages when I go back to chat with the user who I closed(deleted) their chat, but the second point in the new requirement noted that we should be able to see their chats

Reply:
oh right, probably we have something like deleted-at and we don't load messages, this should be changed then, it's more complicated changes needed, we need to see how deactivated method works in status-go

Reply:
@flexsurfer Updated the `DeactivateChat` method from `status-go`.

Link to that OR here: https://github.com/status-im/status-go/pull/4324